modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -74,32 +109,107 @@ def set_module_options(mod)
   end
 
   def is_module_wanted?(mod)
-    (
-      mod.kind_of?(Msf::Exploit::Local) &&
-      mod.has_check? &&
-      is_session_type_compat?(mod) &&
-      is_module_platform?(mod) &&
-      is_module_arch?(mod) &&
+    return false unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
+
+    is_session_type_compat?(mod) &&
+      (@validate_platform ? is_module_platform?(mod) : true) &&
+      (@validate_arch ? is_module_arch?(mod) : true) &&
       is_module_options_ready?(mod)
-    )
   end
 
   def setup
     return unless session
 
     print_status "Collecting local exploits for #{session.session_type}..."
 
+    setup_validation_options
+    setup_color_options
+    setup_tables
+
     # Collects exploits into an array
     @local_exploits = []
-    framework.exploits.each do |name, _obj|
+    framework.exploits.each_with_index do |exploit, index|
+      @valid_reasons = []
+      name = exploit[0]
+
+      print "%bld%blu[*]%clr Gathering exploit #{index + 1} / #{framework.exploits.count}\r"
       mod = framework.exploits.create name
       next unless mod
+
       set_module_options mod
-      next unless is_module_wanted? mod
+
+      module_wanted = is_module_wanted? mod
+
+      add_invalid_module(mod) if @valid_reasons.any?
+
+      next unless module_wanted
+
       @local_exploits << mod
     end
   end
 
+  def setup_validation_options
+    @validate_arch = datastore['ValidateArch']
+    @validate_platform = datastore['ValidatePlatform']
+    @validate_meterpreter_commands = datastore['ValidateMeterpreterCommands']
+  end
+
+  def setup_color_options
+    @valid_color, @invalid_color, @ignored_color =
+      (datastore['Colors'] || '').split('/')
+
+    @valid_color = "%#{@valid_color}" unless @valid_color.blank?
+    @invalid_color = "%#{@invalid_color}" unless @invalid_color.blank?
+    @ignored_color = "%#{@ignored_color}" unless @ignored_color.blank?
+  end
+
+  def setup_stylers
+    @valid_modules_name_styler = ::Msf::Ui::Console::TablePrint::CustomColorStyler.new({})
```

Comment:
If we leave the presentational logic to the end, then we wouldn't need to bind these as instance variables either, we'd just be able to have it as local variables 👀